File: sento-sample/src/main/java/io/sento/sample/annotations/OnFocusReceived.java
package io.sento.sample.annotations;

import io.sento.ListenerBinding;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.CLASS)
@Target(ElementType.METHOD)
@ListenerBinding(
    owner = "android.view.View",
    listener = "io.sento.sample.annotations.OnFocusReceivedListener",
    setter = "setOnFocusChangeListener"
)
public @interface OnFocusReceived {
  public int[] value();
}

File: sento-sample/src/main/java/io/sento/sample/annotations/OnFocusReceived.kt
package io.sento.sample.annotations

import android.view.View
import io.sento.annotations.ListenerClass

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@ListenerClass(
    owner = "android.view.View",
    listener = "io.sento.sample.annotations.OnFocusReceivedListener",
    setter = "setOnFocusChangeListener",
    callback = "onFocusReceived"
)
public annotation class OnFocusReceived(vararg val value: Int)

public abstract class OnFocusReceivedListener : View.OnFocusChangeListener {
  override fun onFocusChange(view: View, focused: Boolean) {
    if (focused) {
      onFocusReceived(view)
    }
  }

  abstract fun onFocusReceived(view: View)
}
